tests/conditional_compilation/conftest.py

```diff
@@ -79,6 +79,33 @@ def pytest_generate_tests(metafunc):
         params.append(pytest.param(test_id, Path(expand_env_vars(model_path)), **extra_args))
         ids = ids + [test_id]
     metafunc.parametrize("test_id, model", params, ids=ids)
+    setattr(metafunc.config, "orig_cases", test_cases)
+
+
+@pytest.fixture(scope="function")
+def test_info(request, pytestconfig):
+    setattr(request.node._request, "test_info", {"orig_instance": pytestconfig.orig_cases,
+                                                 "csv_model_path": {}
+                                                 })
+    if not hasattr(pytestconfig, "session_info"):
+        setattr(pytestconfig, "session_info", [])
+
+    yield request.node._request.test_info
+
+    pytestconfig.session_info.append(request.node._request.test_info)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def update_test_conf_info(pytestconfig, metafunc):
+    yield
+    csv_model_path = pytestconfig.getoption('csv_model_path')
+    if csv_model_path:
+        upd_cases = pytestconfig.orig_cases.copy()
+        for record in pytestconfig.session_info:
+            rec_i = upd_cases.index(record["orig_instance"])
```

Comment:
pytest supports equal test cases, so there is a side effect with searching of the test case